hivemind/dht/__init__.py

```diff
@@ -254,6 +263,24 @@ def add_validators(self, record_validators: Iterable[RecordValidatorBase]) -> No
     async def _add_validators(self, node: DHTNode, record_validators: Iterable[RecordValidatorBase]) -> None:
         node.protocol.record_validator.extend(record_validators)
 
+    @property
+    def peer_id(self) -> PeerID:
+        if self._peer_id is None:
+            self._peer_id = self.run_coroutine(DHT._get_peer_id)
+        return self._peer_id
+
+    async def _get_peer_id(self, node: DHTNode) -> PeerID:
+        return node.peer_id
+
+    @property
+    def client_mode(self) -> bool:
+        if self._client_mode is None:
+            self._client_mode = self.run_coroutine(DHT._get_client_mode)
```

Comment:
By the way, can't we just call `await self._node.protocol.client_mode`? Is that because they're in different processes?

Reply:
That's true, `self._node` exists in the child process but not in the parent one.